src/metatrain/experimental/gap/model.py

```diff
@@ -201,8 +207,25 @@ def forward(
         soap_features = TorchTensorMap(self._keys, soap_features.blocks())
         output_key = list(outputs.keys())[0]
         energies = self._subset_of_regressors_torch(soap_features)
-        out_tensor = self.apply_composition_weights(systems, energies)
-        return {output_key: out_tensor}
+        return_dict = {output_key: energies}
+
+        # apply composition model
+        composition_energies = self.composition_model(
+            systems, {output_key: ModelOutput("energy", per_atom=True)}
+        )
+        # apply selected_atoms to the composition energies if needed
+        if selected_atoms is not None:
+            composition_energies[output_key] = metatensor.torch.slice(
+                composition_energies[output_key], "samples", selected_atoms
+            )
```

Comment:
I think we can also move this inside the CompositionModel itself?

there is one solution in the LJ example of Guillaume.

I recreated the system in a code.

```python
            # select only real atoms and discard ghosts
            if selected_atoms is not None:
                current_system_mask = selected_atoms.column("system") == system_i
                current_atoms = selected_atoms.column("atom")
                current_atoms = current_atoms[current_system_mask].to(torch.long)

                system_final = System(
                    types=system.types[current_atoms],
                    positions=system.positions[current_atoms],
                    cell=system.cell,
                )
```

Reply:
Thanks, I did it in a slightly different way to avoid label conflicts but it's now there